sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -206,6 +206,18 @@ trait CodegenSupport extends SparkPlan {
   def doConsume(ctx: CodegenContext, input: Seq[ExprCode], row: ExprCode): String = {
     throw new UnsupportedOperationException
   }
+
+  /**
+   * for optimization to suppress shouldStop() in a loop of WholeStageCodegen
+   *
+   * isShouldStopRequired: require to insert shouldStop() into the loop if true
+   */
+  def isShouldStopRequired: Boolean = {
+    return shouldStopRequired && !(this.parent != null && !this.parent.isShouldStopRequired)
+  }
+
+  // set false if doConsume() does not insert append() that requires shouldStop() in the loop
```

Comment:
Suggestion for the comment:

    /**
     * Set to false if this plan consumes all rows produced by children but doesn't output row to buffer
     * by calling `append()`, so the children don't require `shouldStop()` in the loop of producing rows.
     */